freqtrade/exchange/exchange.py

```diff
@@ -576,7 +617,21 @@ def get_min_pair_stake_amount(self, pair: str, price: float,
         # The value returned should satisfy both limits: for amount (base currency) and
         # for cost (quote, stake currency), so max() is used here.
         # See also #2575 at github.
-        return max(min_stake_amounts) * amount_reserve_percent
+        return self._apply_leverage_to_stake_amount(
+            max(min_stake_amounts) * amount_reserve_percent,
+            leverage or 1.0
+        )
+
+    def _apply_leverage_to_stake_amount(self, stake_amount: float, leverage: float):
+        """
+        #TODO-lev: Find out how this works on Kraken and FTX
+        # * Should be implemented by child classes if leverage affects the stake_amount
+        Takes the minimum stake amount for a pair with no leverage and returns the minimum
+        stake amount when leverage is considered
+        :param stake_amount: The stake amount for a pair before leverage is considered
+        :param leverage: The amount of leverage being used on the current trade
+        """
+        return stake_amount / leverage
```

Comment:
correct me if i'm wrong - but isn't this the reverse to what we discussed lately?

looking at the place it's used it seems correct - however the naming is highly missleading, as (purely from the name) i've assumed this would have "stake_amount=500$" and leverage=2 return 1000$ - but it's actually doing the reverse.

As said, it's fine for where it's used, but maybe naming of the method can be different / better?